### PR TITLE
DEPLOY-546 CentOS 7.5; Java 8u181; Use --build-arg for Java file; Simpler Build Script

### PR DESCRIPTION
**Pull request by nicdoye**
Branch built via new method:
~~https://bamboo.alfresco.com/bamboo/browse/PS-DBJIR0-JOB1-1/log~~
~~https://bamboo.alfresco.com/bamboo/browse/PS-DBJIR0-JOB1-3/log~~
~~https://quay.io/repository/alfresco/alfresco-base-java/manifest/sha256:01ef6db3244a719690c968b2a6595412f2fb9c8597cd9fe715fe7c752739034e~~

https://bamboo.alfresco.com/bamboo/browse/PS-DBJIR0-JOB1-4/log
https://quay.io/repository/alfresco/alfresco-base-java/manifest/sha256:7625782b23de120f04eb285a0cc796a97403e218a8507079d7aef9812d5df8b1

This is a stepping stone on the way to
a) simpler builds
2) multiple Java versions
iii) de-bambooification

### CODE REVIEW

**Comment by nicdoye** [comment]
Also bumping base versions....

**Comment by derekhulley** [comment]
Please update the PR and add a bit of elaboration e.g. DEPLOY-123: Update blah blah.
Otherwise I have to go to JIRA to see what it is or look at the diff.  I can do that but it takes time and it's not that inclusive for anyone else looking at the commit log.

File: build-prep.sh [hunk 1]
```diff
@@ -26,26 +27,55 @@ if [ "${USE_MVN}" = 'true' ]; then
...
-else
+    JRE_FILENAME="${DOWNLOAD_DIR}/${JAVA_SE_TYPE}-${JAVA_VERSION}-bin.${JAVA_PACKAGING}"
+}
 
+java::download::oracle () {
     # You can still download the latest Java version from Oracle
 
+    declare OLD_PWD="${PWD}"
+    cd "${DOWNLOAD_DIR}"
     curl -jksSLOH "Cookie: oraclelicense=accept-securebackup-cookie" "${JRE_URL}"
```
Comment (kconkas) [inline build-prep.sh]: What happens if `curl` doesn't manage to download the file? Are there any checks for file existence thereafter?

Comment (rgauss) [reply build-prep.sh]: The `errexit` line should cause the whole script to fail in that case.

Comment (nicdoye) [reply build-prep.sh]: Unlike most things I write, this has `set -e` at the top, so the script would exit immediately. The `sha256sum` checks below would catch anything else.

Comment (kconkas) [reply build-prep.sh]: Cool. I should've paid more attention to the beginning of the script. :-)

**Comment by nicdoye** [comment]
@rgauss - it's like a (UK) government bill, where they sneak through extra legislation unrelated to the title... But yes.

**Pull request merged**